luarules/gadgets/unit_liche_cruisecontrol.lua

```diff
@@ -0,0 +1,30 @@
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+function gadget:GetInfo()
+	return {
+		name = "Liche Cruise Control",
+		desc = "Prevent Liches diving when attacking",
+		author = "Hornet, Robert",
+		date = "March 1st, 2024",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+
+
+local licheId = UnitDefNames.armliche.id
```

Comment:
As with the infestor code, even though `armliche` is unlikely to be undefined it's best to be certain.
```
local licheId 
local liche = UnitDefNames.armliche
if liche then 
    licheId = liche .id 
else 
    gadgetHandler:RemoveGadget(self) 
end
```

Reply:
I'd solved this in both but the code isn't here yet (perfi being slightly busy in the grand melee yesterday); the removegadget is a nice idea, will integrate